core/pubnub_grant_token_api.c

```diff
@@ -99,13 +100,32 @@ pubnub_chamebl_t pubnub_get_grant_token(pubnub_t* pb)
     return result;
 }
 
+static unsigned int safe_alloc_strcat(char* destination, const char* source, unsigned int current_allocation_size) {
+    unsigned int concat_size = strlen(destination) + strlen(source);
+    if (concat_size > current_allocation_size) {
+	unsigned int new_allocation_size = (5 * current_allocation_size) / 4; // +20%
+	char* new_alloc = (char*)malloc(new_allocation_size);
```

Comment:
just checking if we need to free "new_alloc" or not because we did malloc?

Reply:
We do not need to free `new_alloc` because it just replacing another allocation. `destination` allocation is freed somewhere else so only what we have to do is to clean old allocation in case of replacement.